sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -58,7 +62,11 @@ abstract class StringRegexExpression extends BinaryExpression
   } else {
     // Let it raise exception if couldn't compile the regex string
     try {
-      Pattern.compile(escape(str))
+      var patternFlags: Int = 0
+      if (CollationFactory.fetchCollation(collationId).supportsLowercaseEquality) {
+        patternFlags = Pattern.CASE_INSENSITIVE
```

Comment:
Same comment as for "u" and "i" flags, we need to add `Pattern.UNICODE_CASE`. Also, consider using `val` instead of `var`.